engine/src/main/java/org/kigalisim/engine/support/RecalcOperationBuilder.java

```diff
@@ -0,0 +1,249 @@
+/**
+ * Builder for creating RecalcOperation instances with fluent interface.
+ *
+ * <p>This builder provides a fluent interface for constructing sequences of
+ * recalculation operations while enforcing proper ordering (recalc methods first,
+ * then propagate methods).</p>
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.kigalisim.engine.support;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.kigalisim.engine.state.Scope;
+
+/**
+ * Builder for creating RecalcOperation instances with fluent interface.
+ */
+public class RecalcOperationBuilder {
+
+  private Scope scopeEffective;
```

Comment:
Please use `Optional` and check `isEmpty` instead of using null for `scopeEffective` and `subtractRecharge`. Please also instantiate these values inside the default constructor to an empty Optional.